1.6.1/docker-entrypoint.sh

```diff
@@ -14,22 +14,52 @@
 set -e
 
 if [ "$1" = 'couchdb' ]; then
-  # we need to set the permissions here because docker mounts volumes as root
-  chown -R couchdb:couchdb \
-    /usr/local/var/lib/couchdb \
-    /usr/local/var/log/couchdb \
-    /usr/local/var/run/couchdb \
-    /usr/local/etc/couchdb
-
-  chmod -R 0770 \
-    /usr/local/var/lib/couchdb \
-    /usr/local/var/log/couchdb \
-    /usr/local/var/run/couchdb \
-    /usr/local/etc/couchdb
-
-  chmod 664 /usr/local/etc/couchdb/*.ini
-  chmod 775 /usr/local/etc/couchdb/*.d
-  exec gosu couchdb "$@"
+	# Start CouchDB
+	couchdb -b > /dev/null 2>&1
+	while ! curl -s 127.0.0.1:5984 > /dev/null; do sleep 1; done
+
+	if [ "$COUCHDB_USER" ] && [ "$COUCHDB_PASSWORD" ]; then
+		# Create user
+		echo "Creating CouchDB admin: \"$COUCHDB_USER\"..."
+		curl -X PUT -sS 127.0.0.1:5984/_config/admins/$COUCHDB_USER -d \"${COUCHDB_PASSWORD}\" > /dev/null 2>&1
```

Comment:
Yeah, I thought of that too. That would eliminate the need to start couch just for PUTting the admin, but we couldn't display the warning below as easily.